A trace here:

<details>

```
odoo_1  | 2018-05-08 12:17:59,655 45 ERROR test odoo.http: Exception during JSON request handling.
odoo_1  | Traceback (most recent call last):
odoo_1  |   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 641, in _handle_exception
odoo_1  |     return super(JsonRequest, self)._handle_exception(exception)
odoo_1  |   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 683, in dispatch
odoo_1  |     result = self._call_function(**self.params)
odoo_1  |   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 333, in _call_function
odoo_1  |     return checked_call(self.db, *args, **kwargs)
odoo_1  |   File "/opt/odoo/custom/src/odoo/odoo/service/model.py", line 101, in wrapper
odoo_1  |     return f(dbname, *args, **kwargs)
odoo_1  |   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 326, in checked_call
odoo_1  |     result = self.endpoint(*a, **kw)
odoo_1  |   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 941, in __call__
odoo_1  |     return self.method(*args, **kw)
odoo_1  |   File "/opt/odoo/custom/src/odoo/odoo/http.py", line 506, in response_wrap
odoo_1  |     response = f(*args, **kw)
odoo_1  |   File "/opt/odoo/auto/addons/mass_mailing_custom_unsubscribe/controllers/main.py", line 113, in unsubscribe
odoo_1  |     mailing._unsubscribe_token(res_id, token)
odoo_1  |   File "/opt/odoo/auto/addons/mass_mailing_custom_unsubscribe/models/mail_mass_mailing.py", line 34, in _unsubscribe_token
odoo_1  |     key = (self.env.cr.dbname, self.id, int(res_id))
odoo_1  | TypeError: int() argument must be a string or a number, not 'NoneType'
```

</details>

--- a/mass_mailing_custom_unsubscribe/models/mail_unsubscription.py
+++ b/mass_mailing_custom_unsubscribe/models/mail_unsubscription.py
@@ -90,8 +90,8 @@ class MailUnsubscription(models.Model):
         """Get the mass mailing list, if it is possible."""
         for one in self:
             try:
-                one.mailing_list_id = one.unsubscriber_id["list_id"]
-            except KeyError:
+                one.mailing_list_id = one.unsubscriber_id.list_id
+            except AttributeError:
                 # Possibly model != mail.mass_mailing.contact; no problem
                 pass
 

--- a/mass_mailing_custom_unsubscribe/models/mail_unsubscription.py
+++ b/mass_mailing_custom_unsubscribe/models/mail_unsubscription.py
@@ -35,6 +35,14 @@ class MailUnsubscription(models.Model):
         lambda self: self._selection_unsubscriber_id(),
         "(Un)subscriber",
         help="Who was subscribed or unsubscribed.")
+    mailing_list_id = fields.Many2one(
+        "mail.mass_mailing.list",
+        "Mailing list",
+        ondelete="set null",
+        compute="_compute_mailing_list_id",
+        store=True,
+        help="(Un)subscribed mass mailing list, if any.",
+    )
     reason_id = fields.Many2one(
         "mail.unsubscription.reason",
         "Reason",
@@ -75,6 +83,17 @@ class MailUnsubscription(models.Model):
             if not one.details and one.details_required:
                 raise exceptions.DetailsRequiredError(
                     _("Please provide details on why you are unsubscribing."))
+
+    @api.multi
+    @api.depends("unsubscriber_id")
+    def _compute_mailing_list_id(self):
+        """Get the mass mailing list, if it is possible."""
+        for one in self:
+            try:
+                one.mailing_list_id = one.unsubscriber_id["list_id"]
+            except KeyError:
+                # Possibly model != mail.mass_mailing.contact; no problem
+                pass
 
     @api.model
     def create(self, vals):

--- a/mass_mailing_custom_unsubscribe/views/mail_unsubscription_view.xml
+++ b/mass_mailing_custom_unsubscribe/views/mail_unsubscription_view.xml
@@ -14,6 +14,7 @@
                     <field name="date"/>
                     <field name="mass_mailing_id"/>
                     <field name="unsubscriber_id"/>
+                    <field name="mailing_list_id"/>
                     <field name="email"/>
                     <field name="action"/>
                     <field name="reason_id"
@@ -43,6 +44,7 @@
             <field name="date"/>
             <field name="mass_mailing_id"/>
             <field name="unsubscriber_id"/>
+            <field name="mailing_list_id"/>
             <field name="email" invisible="True"/>
             <field name="action"/>
             <field name="reason_id"/>
@@ -58,6 +60,7 @@
         <search>
             <field name="mass_mailing_id"/>
             <field name="unsubscriber_id"/>
+            <field name="mailing_list_id"/>
             <field name="email"/>
             <field name="reason_id"/>
             <field name="details"/>
@@ -85,8 +88,9 @@
     <field name="model">mail.unsubscription</field>
     <field name="arch" type="xml">
         <pivot string="(Un)subscriptions">
-            <field name="action" type="row"/>
             <field name="reason_id" type="row"/>
+            <field name="mailing_list_id" type="row"/>
+            <field name="action" type="col"/>
         </pivot>
     </field>
 </record>

